code/mob/living/critter/sawfly.dm

```diff
@@ -0,0 +1,219 @@
+		if(fliesnearby<3) beepchance -=20 //heavily reduce chance of beep in swarm
+		if(prob(beepchance))
+			if(isalive(src))
+				playsound(src, pick(src.beeps), 40, 1)
+				src.visible_message("<b>[src] [beeptext].</b>")
+
+
+
+	emp_act() //same thing as if you emagged the controller, but much higher chance
+		if(prob(80))
+			src.visible_message("<span class='combat'>[src] buzzes oddly and starts to spiral out of control!</span>")
+			SPAWN(2 SECONDS)
+				src.blowup()
+		else
+			src.foldself()
+
+
+	Cross(atom/movable/mover) //code that ensures projectiles hit them when they're alive, but won't when they're dead
+		if(istype(mover, /obj/projectile))
+			if(!isalive(src))
+				return 1
+		return ..()
+
+//note: due to the AIholder's timed nature, they can still priority attack you if you're already targeted, but it's incredibly rare. Frankly I think it adds to the challenge.
+	attackby(obj/item/W as obj, mob/living/user as mob)
+		if(!(istraitor(user) || isnukeop(user) || isspythief(user) || (user in src.friends)))//are you an eligible target?
+			if(prob(50) && isalive(src))//can we attack?
+				if((ai.target != user)) //are we already attacking?
+					ai.interrupt()//stop what you're doing!
+					src.visible_message("<span class='alert'><b>[src]'s targeting subsystems identify [user] as a high priority threat!</b></span>")
+					Shoot(get_turf(user), src.loc, src) //getting this to work in the AIholder was a pain in the butt so I'm moving it here
+					SPAWN(5)
+						Shoot(get_turf(user), src.loc, src)
+
+		..()
+
+	death()
+		if(!isalive(src)) return//we already dead, somehow
+
+		src.set_density(0)
+		src.set_a_intent(INTENT_HELP)
+		src.force_laydown_standup()
+		src.tokenized_message(death_text)
+		src.is_npc = FALSE // //shut down the AI
+		src.throws_can_hit_me = FALSE  //prevent getting hit by thrown stuff- important in avoiding jank
+
+
+		animate(src) //stop no more float animation
+		icon_state = "sawflydead[pick("1", "2", "3", "4", "5", "6", "7", "8")]" //randomly selects death icon and displaces them
+		src.pixel_x += rand(-5, 5)
+		src.pixel_y += rand(-1, 5)
+		src.anchored = 0
+		//walk_to(src,0) //halt walking
+
+
+
+		// special checks that determine how much damage they do after death
+		if (prob(20))
+			new /obj/item/device/prox_sensor(src.loc)
+			return
+		if(prob(60))
+			elecflash(src, 1, 3)
+
+		if(prob(20)) // congrats, little guy! You're special! You're going to blow up!
+			if(prob(70)) //decide whether or not people get a warning
+				src.visible_message("<span class='combat'>[src] makes /a [pick("gentle", "odd", "slight", "weird", "barely audible", "concerning", "quiet")] [pick("hiss", "drone", "whir", "thump", "grinding sound", "creak", "buzz", "khunk")]...")
+			SPAWN(deathtimer SECONDS)
+				src.blowup()
+
+
+		/*remove_lifeprocess(/datum/lifeprocess/blood) //so apparently the reduce_lifeprocess_on_death() proc is limited to only the small animals
```

Comment:
Weird, does it still do the density resetting thing though?

Reply:
tested this a whole bunch, and yes, that there line is vital to them having no density on death